### PR TITLE
Add remove_spaces to handle sequences with spaces

### PR DESCRIPTION
**Pull request by mataton**
This PR addresses issue #2148. ~~It is WIP for now because of some unexpected behavior in one of the doctests and possibly adding more unit tests. However,~~ the current code should be functional.

Please complete the following checklist:

* [x] I have read the [contribution guidelines](https://scikit.bio/contribute.html).

* [x] I have documented all public-facing changes in the [changelog](https://github.com/scikit-bio/scikit-bio/blob/main/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/scikit-bio/scikit-bio/blob/main/LICENSE.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied.

  - **It is your responsibility to disclose** code, documentation, or other content derived from external source(s). If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] This pull request does not include code, documentation, or other content derived from external source(s).

Note: [This document](https://scikit.bio/devdoc/review.html) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

**Review by qiyunzhu — COMMENTED** [review]
Hi @mataton This is great job. But I after thinking harder I wanted propose a concern: The removal of spaces takes place during constructing a `Sequence` instance. This implies that if one wants to convert one `Sequence` into another `Sequence`, the existing spaces (if any) will be lost. I am not sure if this is the desired behavior. To my knowledge, spaces in the sequence happens in the FASTA, GenBank and Phylip formats, but they don't happen in the FASTQ format. I wonder if the process should take place in the FASTA parser instead? Meanwhile, I wonder if the GenBank and Phylip parsers in scikit-bio already have this mechanism? Finally, `remove_spaces` becomes an attribute of `self` and never gets cleared. This is a minor issue.

**Comment by qiyunzhu** [comment]
For example, the phylip parser has [this](https://github.com/scikit-bio/scikit-bio/blob/d7973bfdd12f083fe9dccc319a3793e829755720/skbio/io/format/phylip.py#L321) @mataton

File: CHANGELOG.md [hunk 1]
```diff
@@ -53,6 +53,7 @@
 * Fixed a bug in `TreeNode.unrooted_move` which does not respect specified branch attributes ([#2103](https://github.com/scikit-bio/scikit-bio/pull/2103)).
 * Fixed a bug in `skbio.diversity.get_beta_diversity_metrics` which does not display metrics other than UniFrac ([#2126](https://github.com/scikit-bio/scikit-bio/pull/2126)).
 * Raises an error when beta diversity metric `mahalanobis` is called but sample number is smaller than or equal to feature number in the data. Thank @quliping for noting this in [#2125](https://github.com/scikit-bio/scikit-bio/pull/2125).
+* Fixed a bug in `skbio.sequence.Sequence` that improperly handled sequences containing spaces. ([#2156](https://github.com/scikit-bio/scikit-bio/pull/2156))
```
Comment (qiyunzhu) [inline CHANGELOG.md:56]: I think the bug is in `io.format.fasta`, not in `sequence.Sequence`?

**Comment by mataton** [comment]
@qiyunzhu Should be ready to merge if all checks pass.

**Comment by mataton** [comment]
@qiyunzhu Can a valid FASTA file have numbers for sequences? For instance:

```bash
$ cat skbio/io/format/tests/data/qual_3_seqs_non_defaults
>s*e*q*1 desc+1
1234
0 0
2
>s*e*q*2 desc+2
1
11
111
11112
>s*e*q*3 desc+3
12345
678909
999999
4242424242
```

This is being treated as a valid FASTA file in our unit tests currently, is this accurate?

**Comment by mataton** [comment]
@qiyunzhu One clarifying question. If FASTA can contain any values then what is the difference between these two files?

`qual_2_seqs_defaults` is considered to NOT be a valid FASTA file in our unit tests.
```bash
$ cat skbio/io/format/tests/data/qual_2_seqs_defaults
>s_e_q_1 desc 1
1 2 3 4
>s_e_q_2          desc 2
 42    41 39 40
```

On the other hand, `qual_3_seqs_non_defaults` IS considered to be a valid FASTA file.
```bash
$ cat skbio/io/format/tests/data/qual_3_seqs_non_defaults
>s*e*q*1 desc+1
1234
0 0
2
>s*e*q*2 desc+2
1
11
111
11112
>s*e*q*3 desc+3
12345
678909
999999
4242424242
```

What is the difference between these two files?

**Comment by qiyunzhu** [comment]
This is good to go!

**Pull request merged**